test/data/prototype/entity/mining_drillTests.cpp

```diff
@@ -150,38 +122,115 @@ namespace jactorio::data
 		// Mining drill is built with an item output chest
 
+		drillProto_.miningSpeed          = 2;  // Halves mining time
 		drillProto_.resourceOutput.right = {3, 1};
 		ContainerEntity container{};
 
 
-		SetupChest(worldData_, container);
+		TestSetupContainer(worldData_, {4, 2}, container);
+		auto& tile = TestSetupDrill(worldData_, logicData_,
+		                            {1, 1},
+		                            resource_, drillProto_, 100);
 
-		SetupDrill(worldData_, logicData_, resource_, drillProto_);
+		auto* data = tile.GetLayer(game::ChunkTile::ChunkLayer::entity).GetUniqueData<MiningDrillData>();
 
+		EXPECT_EQ(data->resourceCoord.x, 0);
+		EXPECT_EQ(data->resourceCoord.y, 0);
+		EXPECT_EQ(data->resourceOffset, 6);
 
 		// ======================================================================
-		// Unique data created by on_build()
-		game::ChunkTile* tile = worldData_.GetTile(1, 1);
+		// Resource taken from ground
+		auto& resource_layer = tile.GetLayer(game::ChunkTile::ChunkLayer::resource);
 
-		game::ChunkTileLayer& layer = tile->GetLayer(game::ChunkTile::ChunkLayer::entity);
-		auto* data                  = layer.GetUniqueData<MiningDrillData>();
+		EXPECT_EQ(resource_layer.GetUniqueData<ResourceEntityData>()->resourceAmount, 99);
 
+		// ======================================================================
 		// Ensure it inserts into the correct entity
+
 		Item item{};
 		data->outputTile.DropOff(logicData_, {&item, 1});
 
-		game::ChunkTileLayer& container_layer = worldData_.GetTile(4, 2)
-		                                                  ->GetLayer(
-			                                                  game::ChunkTile::ChunkLayer::entity);
+		game::ChunkTileLayer& container_layer =
+			worldData_.GetTile(4, 2)->GetLayer(game::ChunkTile::ChunkLayer::entity);
 
-		EXPECT_EQ(container_layer.GetUniqueData<jactorio::data::ContainerEntityData>()->inventory[0].count,
-		          1);
+		EXPECT_EQ(container_layer.GetUniqueData<ContainerEntityData>()->inventory[0].count, 1);
 
 		// ======================================================================
-		logicData_.deferralTimer.DeferralUpdate(worldData_, 60);  // Takes 60 ticks to mine
 
-		EXPECT_EQ(container_layer.GetUniqueData<jactorio::data::ContainerEntityData>()->inventory[1].count,
-		          1);
+		logicData_.deferralTimer.DeferralUpdate(worldData_, 30);  // Takes 60 ticks to mine / 2 (since mining speed is 2)
+
+		EXPECT_EQ(container_layer.GetUniqueData<ContainerEntityData>()->inventory[1].count, 1);
+
+		// Another resource taken for next output
+		EXPECT_EQ(resource_layer.GetUniqueData<ResourceEntityData>()->resourceAmount, 98);
+	}
+
+	TEST_F(MiningDrillTest, ExtractRemoveResourceEntity) {
+		drillProto_.resourceOutput.right = {3, 1};
+		ContainerEntity container{};
+
+
+		TestSetupContainer(worldData_, {4, 2}, container);
+
+		// Second resource must be set up prior to the drill, since it will remove the first existing resource and
```

Comment:
This is not true, it will search for another resource during the DeferralUpdate callback